pkg/types/vsphere/machinepool_test.go

```diff
@@ -0,0 +1,99 @@
+package vsphere
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestSet_PreservesZonesFromDefaultMachinePlatform tests that zones from defaultMachinePlatform
+// are preserved when pool-specific platform config is set without zones.
+// This reproduces the bug reported in OCPBUGS-62209.
```

Comment:
:question: I think test cases in this file are testing the `Set` func, right?

https://github.com/openshift/installer/blob/4180662515e75d11d0a7984c66d574f3c64a765a/pkg/types/vsphere/machinepool.go#L94

I am just unsure it is "truly" reproduce [OCPBUGS-62209](https://issues.redhat.com//browse/OCPBUGS-62209), where zones are defaulted in validation (where it should not be). I guess we can adjust the comments (golint is also failing here :grin:)?